### PR TITLE
feat: [AsPu-703, AsPu-696, AsPu-710] improved accessibility skipping to main content

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
**YouTrack:**
- [Course Navigation - Dates page](https://youtrack.raccoongang.com/issue/AsPu-710/Course-Navigation-Dates-page)
- [Course Navigation - Course unit](https://youtrack.raccoongang.com/issue/AsPu-696/Course-Navigation-Course-unit)
- [Course Navigation - Progress page](https://youtrack.raccoongang.com/issue/AsPu-703/Course-Navigation-Progress-page)

**Behavior example:**
image

### CODE REVIEW

File: src/generic/hooks.js [hunk 1]
```diff
@@ -59,3 +59,86 @@ export function useIFrameHeight(onIframeLoaded = null) {
...
+  useEffect(() => {
+    const skipLinkElement = document.querySelector(skipLinkSelector);
+    skipLinkElementRef.current = skipLinkElement;
+
+    if (skipLinkElement) {
+      skipLinkElement.addEventListener('keydown', handleSkipAction);
+      skipLinkElement.addEventListener('click', handleSkipAction);
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn(`Skip link with selector "${skipLinkSelector}" not found.`);
```
Comment (vladislavkeblysh) [inline src/generic/hooks.js:132]: Do we need this console.warn?

File: src/generic/hooks.js [hunk 2]
```diff
@@ -59,3 +59,86 @@ export function useIFrameHeight(onIframeLoaded = null) {
...
+   */
+  const handleSkipAction = useCallback((event) => {
+    if (shouldTriggerSkip(event)) {
+      event.preventDefault();
+      const targetElement = document.getElementById(targetElementId);
+      if (targetElement) {
+        scrollToTarget(targetElement);
+      } else {
+        // eslint-disable-next-line no-console
+        console.warn(`Element with ID "${targetElementId}" not found.`);
```
Comment (vladislavkeblysh) [inline src/generic/hooks.js:118]: Do we need this console.warn?

File: src/generic/hooks.js [hunk 3]
```diff
@@ -59,3 +59,86 @@ export function useIFrameHeight(onIframeLoaded = null) {
...
+   */
+  const scrollToTarget = (targetElement) => {
+    const targetPosition = targetElement.getBoundingClientRect().top + window.scrollY;
+    window.scrollTo({ top: targetPosition - scrollOffset, behavior: 'smooth' });
+
+    if (typeof targetElement.focus === 'function') {
+      targetElement.focus({ preventScroll: true });
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn(`Element with ID "${targetElementId}" exists but is not focusable.`);
```
Comment (vladislavkeblysh) [inline src/generic/hooks.js:93]: Do we need this console.warn?

File: src/generic/hooks.js (hunk 1 above)
Comment (peterkulko) [reply src/generic/hooks.js:132]: Yes, these are logs for developers.

File: src/generic/hooks.js (hunk 2 above)
Comment (peterkulko) [reply src/generic/hooks.js:118]: Yes, these are logs for developers.

File: src/generic/hooks.js (hunk 3 above)
Comment (peterkulko) [reply src/generic/hooks.js:93]: Yes, these are logs for developers.

**Review by ihor-romaniuk — APPROVED** [review]

**Pull request merged**